docs/guide/supervisor-programming.md

```diff
@@ -48,6 +50,112 @@ int main() {
   return 0;
 }
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <webots/Supervisor.hpp>
+
+using namespace webots;
+
+int main(int argc, char **argv) {
+  // create a supervisor instance.
+  Supervisor *supervisor = new Supervisor();
+
+  // get the time step of the current world.
+  int timeStep = (int)supervisor->getBasicTimeStep();
+
+  // retrieve concerned robot node from supervisor
+  Node *robot = supervisor->getFromDef("MY_ROBOT");
+
+  // retrieve concerned field from robot node
+  Field *translation = robot->getField("translation");
+
+  while (supervisor->step(timeStep) != -1) {
+    // retrieve and print values from translation field repeatedly
+    const double *values = translation->getSFVec3f();
```

Comment:
It's a pity you changed the variables names across languages (`trans`, `values`, `res`, etc.) this makes the correspondence difficult to understand. Also `TIME_STEP` is sometimes hard-coded to 32 and sometimes defined in a variable with different spellings... All this should be more consistent. Can you fix it?

Reply:
Do we want to set it manually or give an example on how to retrieve it then ?